web/webkit/src/main/scala/net/liftweb/http/LAFutureWithSession.scala

```diff
@@ -0,0 +1,48 @@
+package net.liftweb.http
+
+import net.liftweb.actor.{LAFuture, LAScheduler}
+import net.liftweb.common.{EmptyBox, Failure, Full}
+
+
+object LAFutureWithSession {
+
+  /**
+    * Creates `LAFuture` instance aware of current request and session. Each `LAFuture` returned by chained
+    * transformation method (e.g. `map`, `flatMap`) will be also request/session-aware. However, it's
+    * important to bear in mind that initial session or request are not propagated to chained methods. It's required
+    * that current execution thread for chained method has its own request or session available if reading/writing
+    * some data to it as a part of chained method execution.
+    */
+  def withCurrentSession[T](task: => T, scheduler: LAScheduler = LAScheduler): LAFuture[T] = {
+    S.session match {
+      case Full(session) =>
+        withSession(task, scheduler)
+
+      case empty: EmptyBox =>
+        withFailure(empty ?~! "LiftSession not available in this thread context", scheduler)
```

Comment:
Don't we normally throw an exception if something like this attempts to access the session without a proper session context? I'd prefer we keep that behavior consistent. It feels like a situation where an exception is more or less appropriate.